src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -1110,6 +1112,59 @@ class spell_mage_prismatic_barrier : public AuraScript
     }
 };
 
+// 205021 - Ray of Frost
+class spell_mage_ray_of_frost : public SpellScript
+{
+    PrepareSpellScript(spell_mage_ray_of_frost);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo ({ SPELL_MAGE_RAY_OF_FROST_FINGERS_OF_FROST });
+    }
+
+    void HandleOnHit()
+    {
+        if (Unit* caster = GetCaster())
+            caster->CastSpell(caster, SPELL_MAGE_RAY_OF_FROST_FINGERS_OF_FROST, TRIGGERED_IGNORE_CAST_IN_PROGRESS);
+    }
+
+    void Register() override
+    {
+        OnHit += SpellHitFn(spell_mage_ray_of_frost::HandleOnHit);
+    }
+};
+
+class spell_mage_ray_of_frost_aura : public AuraScript
+{
+    PrepareAuraScript(spell_mage_ray_of_frost_aura);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo ({ SPELL_MAGE_RAY_OF_FROST_BONUS, SPELL_MAGE_RAY_OF_FROST_FINGERS_OF_FROST });
+    }
+
+    void HandleEffectPeriodic(AuraEffect const* aurEff)
+    {
+        if (Unit* caster = GetCaster())
+        {
+            if (aurEff->GetTickNumber() > 1) // First tick should deal base damage
```

Comment:
aren't you like 1 tick too late? i mean when the 2nd tick happenend, only then you apply the bonus

Reply:
the tick number is increased before the hook is called, so its 1 for the first tick - not 0.
Ref:
https://github.com/TrinityCore/TrinityCore/blob/2a5329847636ddc080c461bda06375edfbf3815d/src/server/game/Spells/Auras/SpellAuraEffects.cpp#L1027-L1036

Reply:
That doesn’t answer my question.. ticksDone starts at 0 and is increased to 1 and then the hook is called and GetTickNumber returns 1 and not 2.

Therefore you still need a second call in order to get this bonus or am I seeing things ?

Reply:
This hook is executes before dmg is done. So the check is fine.

Ref
https://github.com/TrinityCore/TrinityCore/blob/2a5329847636ddc080c461bda06375edfbf3815d/src/server/game/Spells/Auras/SpellAuraEffects.cpp#L1063